src/walmart/WalmartProductList.js

```diff
@@ -0,0 +1,56 @@
+const WalmartProduct = require('./WalmartProduct');
+const fs = require('fs');
+
+/* Class representing a list of walmart items */
+class WalmartProductList {
+
+  constructor(productsJSON) {
+    this.products = this.productsList(productsJSON);
+  }
+
+  // Create a list of walmart products that all have a UPC code.
+  productsList(productsJSON) {
+    let products = [];
+
+    productsJSON.forEach(function(product) {
+      if (product.hasOwnProperty('upc')) {
+        products.push(new WalmartProduct(product));
+      }
+    });
+    return products;
+  }
+
+  // Merge a list of walmart products with this.products
+  addProducts(otherProducts) {
+    this.products = this.products.concat(otherProducts);
+  }
+
+  // Save all products info to a file.
+  writeToFile(fileName) {
+    let special_feeds_file = fs.createWriteStream(fileName);
+    special_feeds_file.on('error', function(err) { console.log(err) });
+    this.products.forEach(function(product) {
+      special_feeds_file.write(product.print());
```

Comment:
For readability, I suggest either using the writeLine() equivalent here or appending a new line in product.print(). There are no new lines at the moment and it's not readable. This applies to all writes you made to files - paginatedFeeds, SpecialFeeds, e.t.c.

Reply:
The new lines actually weren't working previously either. I had to explicitly add the carriage return and new line characters to get it to work. https://github.com/sulaimonlasisi/clearance-app/pull/9/commits/217bc3634f7ed9008a80aa97c3e29132d05430b8

Reply:
Interesting. It was working on my Windows PC. Your fix works now too.